src/pages/workspaces/workspace/Data.test.ts

```diff
@@ -0,0 +1,137 @@
+import { DeepPartial } from '@terra-ui-packages/core-utils';
+import { act, render, screen } from '@testing-library/react';
+import { h } from 'react-hyperscript-helpers';
+import { Ajax } from 'src/libs/ajax';
+import { LeoAppStatus, ListAppResponse } from 'src/libs/ajax/leonardo/models/app-models';
+import { WorkspaceWrapper } from 'src/libs/workspace-utils';
+import { testOnlyExports } from 'src/pages/workspaces/workspace/Data';
+import { asMockedFn } from 'src/testing/test-utils';
+import { defaultAzureWorkspace } from 'src/testing/workspace-fixtures';
+
+import { StorageDetails } from './useWorkspace';
+
+const { WorkspaceData } = testOnlyExports;
+
+type WorkspaceContainerExports = typeof import('src/pages/workspaces/workspace/WorkspaceContainer');
+jest.mock('src/pages/workspaces/workspace/WorkspaceContainer', (): WorkspaceContainerExports => {
+  return {
+    ...jest.requireActual<WorkspaceContainerExports>('src/pages/workspaces/workspace/WorkspaceContainer'),
+    wrapWorkspace: jest.fn().mockImplementation((_opts) => (wrappedComponent) => wrappedComponent),
+  };
+});
+
+type AjaxExports = typeof import('src/libs/ajax');
+jest.mock('src/libs/ajax', (): AjaxExports => {
+  return {
+    ...jest.requireActual<AjaxExports>('src/libs/ajax'),
+    Ajax: jest.fn(),
+  };
+});
+
+describe('WorkspaceData', () => {
+  // When Data.js is broken apart and the WorkspaceData component is converted to TypeScript,
+  // this type belongs there.
+  interface WorkspaceDataProps {
```

Comment:
Style nit: suggest moving this type and AjaxContract to the module level instead of inside the `describe`.

Reply:
Done.  Although I added two more types for the `SetupOptions` and `SetupResult` right in this spot.  LMK if they should get the same treatment.